ghcide/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -137,48 +139,49 @@ runLanguageServer options inH outH getHieDbLoc defaultConfig onConfigurationChan
             registerIdeConfiguration (shakeExtras ide) initConfig
 
             let handleServerException (Left e) = do
-                    logError (ideLogger ide) $
+                    logError logger $
```

Comment:
Should we be sending a message here too?

Reply:
No, since that will result in double notifications being sent, notice the `finally` in `checkCancelled`. 

I'll take the chance to clean it up

Reply:
Nevermind, `checkCancelled` doesn't propagate exceptions, so we should be sending a message here too indeed